packages/scaffold/src/views/w3m-receive-view/index.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  AccountController,
+  AssetUtil,
+  CoreHelperUtil,
+  ModalController,
+  NetworkController,
+  RouterController,
+  SnackController,
+  ThemeController
+} from '@web3modal/core'
+import { UiHelperUtil, customElement } from '@web3modal/ui'
+import { LitElement, html } from 'lit'
+import styles from './styles.js'
+import { state } from 'lit/decorators.js'
+
+@customElement('w3m-receive-view')
+export class W3mReceiveView extends LitElement {
+  public static override styles = styles
+
+  // -- Members ------------------------------------------- //
+  private unsubscribe: (() => void)[] = []
+
+  // -- State & Properties -------------------------------- //
+  @state() private address = AccountController.state.address
+
+  @state() private profileName = AccountController.state.profileName
+
+  @state() private network = NetworkController.state.caipNetwork
+
+  public constructor() {
+    super()
+    this.unsubscribe.push(
+      ...[
+        AccountController.subscribe(val => {
+          if (val.address) {
+            this.address = val.address
+            this.profileName = val.profileName
+          } else {
+            ModalController.close()
```

Comment:
I think instead of closing when we have no address for whatever the reason, we should show a message? Or maybe close but show the message else it'd be confusing imo

Reply:
Added error message